src/meta/meta_ckpt.c

```diff
@@ -590,6 +590,10 @@ __wt_meta_ckptlist_set(
     WT_DECL_RET;
     bool has_lsn;
 
+    /* Review the checkpoint's write generation. */
```

Comment:
@keithbostic 
Ok, the culprit is here. The `__ckptlist_review_write_gen` function is where we intercept if we're looking at the metadata checkpoint and set it to the current `WT_CONNECTION::max_write_gen` value. This needs to be done BEFORE updating the metadata or otherwise when we restart, we will read the previous checkpoint's max write gen.